src/test/java/qna/domain/QuestionTest.java

```diff
@@ -1,6 +1,123 @@
+                Arguments.of(new TestQuestion(Q1), UserTest.JAVAJIGI),
+                Arguments.of(new TestQuestion(Q2), UserTest.SANJIGI)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("questionHasOtherOwnerAnswer")
+    public void delete_다른_사람이_쓴_글(Question question, User loginUser) throws CannotDeleteException {
+        assertThatExceptionOfType(CannotDeleteException.class)
+                .isThrownBy(() -> question.delete(loginUser))
+                .withMessage("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+    }
+
+    private static Stream<Arguments> questionHasOtherOwnerAnswer() {
+        Q1.addAnswer(AnswerTest.A1);
+        Q1.addAnswer(AnswerTest.A2);
+
+        Q2.addAnswer(AnswerTest.A1);
+        Q2.addAnswer(AnswerTest.A2);
+
+        return Stream.of(
+                Arguments.of(Q1, UserTest.JAVAJIGI),
+                Arguments.of(Q2, UserTest.SANJIGI)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("questionSameOwnerAnswer")
+    public void delete_성공_질문자_답변자_같음(TestQuestion testQuestion, User loginUser) throws CannotDeleteException {
+        assertAll(
+                () -> assertThatCode(() -> testQuestion.delete(loginUser))
+                        .doesNotThrowAnyException(),
+                () -> assertThat(testQuestion.isDeleted()).isTrue()
+        );
+    }
+
+    private static Stream<Arguments> questionSameOwnerAnswer() {
+        TestQuestion testQuestion1 = new TestQuestion(Q1);
+        testQuestion1.addAnswer(AnswerTest.A1);
+
+        TestQuestion testQuestion2 = new TestQuestion(Q2);
+        testQuestion2.addAnswer(AnswerTest.A2);
+
+        return Stream.of(
+                Arguments.of(testQuestion1, UserTest.JAVAJIGI),
+                Arguments.of(testQuestion2, UserTest.SANJIGI)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("questionOtherOwner")
+    public void delete_답변_중_다른_사람이_쓴_글() throws CannotDeleteException {
+        User invalidLoginUser = UserTest.SANJIGI;
+        Q1.addAnswer(AnswerTest.A1);
+        Q1.addAnswer(AnswerTest.A2);
+
+        assertThatExceptionOfType(CannotDeleteException.class)
+                .isThrownBy(() -> Q1.delete(invalidLoginUser))
+                .withMessage("질문을 삭제할 권한이 없습니다.");
+    }
+
+    private static Stream<Arguments> questionOtherOwner() {
+        return Stream.of(
+                Arguments.of(Q1, UserTest.SANJIGI),
+                Arguments.of(Q2, UserTest.JAVAJIGI)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("deleteHistories")
+    @DisplayName("반환된 deleteHistories 중 Question DeleteHistory가 포함된다.")
+    public void deleteHistoriesTest(Question question, User loginUser) {
+        DeleteHistories deleteHistories = question.delete(loginUser);
+        List<DeleteHistory> collect = StreamSupport.stream(deleteHistories.spliterator(), false)
+                .collect(Collectors.toList());
+        assertThat(collect).contains(new DeleteHistory(question, LocalDateTime.now()));
```

Comment:
```suggestion
        assertIterableEquals(deleteHistories, Arrays.asList(new DeleteHistory(question, LocalDateTime.now())));
```
이부분이 질문 주신 부분 같은데요 ! 😀 시윤님이 말씀하신 방법도 충분해 보입니다 👍
아니면 위에 처럼 Assertions에서 제공해주는 assertIterableEquals()을 사용해 
Iterable객체간의 비교로 검증하는 방법도 괜찮을 것 같아요 🙂

https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertIterableEquals-java.lang.Iterable-java.lang.Iterable-